lib/logflare/context_cache.ex

```diff
@@ -32,43 +41,51 @@ defmodule Logflare.ContextCache do
         value
 
       {:ok, {:cached, value}} ->
-        index_keys(context, cache_key, value)
+        # already cached don't re-index
         value
     end
   end
 
-  def bust_keys(values) do
-    {:ok, keys} = Cachex.keys(@cache)
+  @doc """
+  This function is called from the CacheBuster process when a new record comes in from the Postgres
+  write-ahead log. The WAL contains records. From those records the CacheBuster picks out
+  primary keys.
 
-    total =
-      Enum.count(keys, fn {token, cache_key} = key ->
-        with true <- token in values,
-             {context, _} = token,
-             context_cache = cache_name(context),
-             {:ok, true} <- Cachex.del(context_cache, cache_key) do
-          Cachex.del(@cache, key)
-        end
+  The records ARE the keys in the reverse cache (the ContextCache).
 
-        true
-      end)
+  We must:
+   - Find the key by the record primary key
+   - Delete the reverse cache entry
+   - Delete the cache entry for that context cache e.g. `Logflare.Users.Cache`
+  """
 
-    :telemetry.execute(
-      [:logflare, :context_cache, :busted],
-      %{count: total},
-      %{}
-    )
+  def bust_keys(values) do
+    {:ok, keys} = Cachex.keys(@cache)
+
+    Enum.each(keys, fn {token, phash_key} = key ->
+      with true <- token in values,
+           {context, _} = token,
+           context_cache = cache_name(context),
+           {:ok, cache_key} = Cachex.get(@cache, key),
+           {:ok, true} <- Cachex.del(context_cache, cache_key) do
+        Cachex.del(@cache, key)
+      end
+    end)
```

Comment:
I wonder if counting how many entries were in fact evicted would be useful. Like:
```suggestion
    busted = Enum.count(keys, fn {token, phash_key} = key ->
      with true <- token in values,
           {context, _} = token,
           context_cache = cache_name(context),
           {:ok, cache_key} = Cachex.get(@cache, key),
           {:ok, true} <- Cachex.del(context_cache, cache_key) do
        Cachex.del(@cache, key)
        
        true
      else
        _ -> false
      end
    end)
    
    :telemetry.execute([:logflare, :context_cache], %{busted: busted}, %{})
```